client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -0,0 +1,578 @@
+	nOpenWindows   atomic.Int32 // number of open windows
+)
+
+const (
+	// macOSAppTitle is the title of the macOS app. It is used to set the title
+	// of the main menu.
+	macOSAppTitle = "Decred DEX"
+	// selOpenLogs is the selector for the "Open Logs" menu item.
+	selOpenLogs = "openLogs:"
+	// selNewWindow is the selector for the "New Window" menu item.
+	selNewWindow = "newWindow:"
+	// selIsNewWebview is the selector for a webview that require a new window.
+	selIsNewWebview = "isNewWebView:"
+)
+
+func init() {
+	runtime.LockOSThread()
+
+	// Set webview preferences.
+	webviewConfig.Preferences().SetValueForKey(core.True, core.String("developerExtrasEnabled"))
+	webviewConfig.Preferences().SetValueForKey(core.True, core.String("javaScriptCanAccessClipboard"))
+	webviewConfig.Preferences().SetValueForKey(core.True, core.String("DOMPasteAllowed"))
+}
+
+// hasOpenWindows is a convenience function to tell if there are any windows
+// currently open.
+func hasOpenWindows() bool {
+	return nOpenWindows.Load() > 0
+}
+
+// mainCore is the darwin entry point for the DEX Desktop client.
+func mainCore() error {
+	appCtx, cancel := context.WithCancel(context.Background())
+	defer cancel() // don't leak on the earliest returns
+
+	// Parse configuration.
+	cfg, err := configure()
+	if err != nil {
+		return fmt.Errorf("configuration error: %w", err)
+	}
+
+	// Return early if unsupported flags are provided.
+	if cfg.Webview != "" {
+		return errors.New("--webview flag is not supported. Other OS use it for a specific reason (to support multiple windows)")
+	}
+
+	// The --kill flag is a backup measure to end a background process (that
+	// presumably has active orders).
+	if cfg.Kill {
+		// This is a desktop app and the interactive quit or force quit
+		// button can be used to shutdown the app.
+		return errors.New(`--kill flag is not supported. Use the "Quit" or "Force Quit" button to kill any running process.`)
+	}
+
+	// Filter registered assets.
+	asset.SetNetwork(cfg.Net)
+
+	// shutdownCloser is used to execute functions that could have been executed
+	// as a deferred statement.
+	shutdownCloser := dex.NewErrorCloser()
+	defer shutdownCloser.Done(log) // execute deferred statements if we return early
+
+	// Use a hidden "dexc-desktop-state" file to prevent other processes when
+	// dexc-desktop is already running (e.g when non-bundled version of
+	// dexc-desktop is executed from cmd and vice versa).
+	dexcDesktopStateFile := filepath.Join(cfg.Config.AppData, cfg.Net.String(), ".dexc-desktop-state")
+	f, err := lockDexcDesktopStateFile(dexcDesktopStateFile)
+	if err != nil {
+		return err
+	}
+	shutdownCloser.Add(func() error {
+		f.Close()
+		return os.Remove(dexcDesktopStateFile)
+	})
+
+	// Initialize logging.
+	utc := !cfg.LocalLogs
+	logMaker, closeLogger := app.InitLogging(cfg.LogPath, cfg.DebugLevel, cfg.LogStdout, utc)
+	shutdownCloser.Add(func() error {
+		log.Debug("Exiting dexc main.")
+		closeLogger()
```

Comment:
> defer closeLogger() right after InitLogging

[Comment](https://github.com/decred/dcrdex/pull/2372/files/4ecd7991a174a2e8705d17328440da7e412bc683#r1224895770)

 > The debug message doesn't fit either.
 
It was in the main function and because this is the last function to be executed, I had to add it here. Will remove it in my next commit.